pdns/dnsdist.cc

```diff
@@ -1247,15 +1247,27 @@ void* maintThread()
 {
   int interval = 1;
   size_t counter = 0;
+  uint16_t secondsToWaitLog = 0;
 
   for(;;) {
     sleep(interval);
 
     {
       std::lock_guard<std::mutex> lock(g_luamutex);
       auto f =g_lua.readVariable<boost::optional<std::function<void()> > >("maintenance");
-      if(f)
-        (*f)();
+      if(f) {
+        try {
+          (*f)();
+          secondsToWaitLog = 0;
+        }
+        catch(LuaContext::ExecutionErrorException &e) {
```

Comment:
Perhaps we could catch `std::runtime_error`, or even `std::exception` instead of `LuaContext::ExecutionErrorException`? Because otherwise we don't catch `LuaContext::SyntaxErrorException` nor `LuaContext::WrongTypeException` for example, although I'm not sure they can be raised in that context.